timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java

```diff
@@ -124,6 +126,13 @@ public static TimeLockAgent create(MetricsManager metricsManager,
                 metricsManager,
                 Suppliers.compose(TimeLockRuntimeConfiguration::paxos, runtime::get));
 
+        // Upgrading the schema version should generally happen AFTER any migration has finished running. Keep this in
+        // mind for any potential live migrations
+        PersistedSchemaVersion persistedSchemaVersion = PersistedSchemaVersion
+                .create(installationContext.sqliteDataSource());
+        persistedSchemaVersion.upgradeVersion(SCHEMA_VERSION);
+        verifySchemaVersion(persistedSchemaVersion);
```

Comment:
Doesn't this need to be before all the setting up of PaxosResources etc.? Isn't this currently where the migration happens?